### PR TITLE
Auth tests

### PR DESCRIPTION
**Pull request by othniel-agera**

### CODE REVIEW

File: test/auth.spec.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        })
+        .set('Accept', 'application/json')
+        .expect(201);
+
+      const resp_data = response.body;
+      expect(resp_data).to.be.an('object');
+      expect(resp_data).to.have.property('accessToken');
+      expect(resp_data).to.have.property('refreshToken');
+      expect(resp_data.accessToken).to.be.an('string');
+      expect(resp_data.refreshToken).to.be.an('string');
```
Comment (KufreOkon) [inline test/auth.spec.js]: Just in addition, you may check that the tokens are not empty strings.

File: test/auth.spec.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        .post('/auth/signup')
+        .send({
+          username: `${Date.now()}_ox`,
+          firstname: 'leo',
+          lastname: 'lenzo',
+          email,
+          password: 'test123',
+        })
+        .set('Accept', 'application/json')
+        .expect(201);
```
Comment (KufreOkon) [inline test/auth.spec.js:41]: For now, it is fine, but later we will see how to handle this dependency using a package called Sinon, so you don't have call the signup API before testing signIn. Also when testing signup, the data is not written to the databse.

File: test/auth.spec.js [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+      username: 'kufre',
+      firstname: 'Kufre',
+      lastname: 'Okon',
+      email: 'kufre@example.com',
+      password: 'test123',
+    };
+    before(async () => {
+      const password = await hashPassword('test123');
+      await createUser({ ...user, password }).id;
+    });
```
Comment (KufreOkon) [inline test/auth.spec.js:73]: I love this 👍

**Review by KufreOkon — CHANGES_REQUESTED** [review]
Good work 👍

File: test/auth.spec.js (hunk 2 above)
Comment (othniel-agera) [reply test/auth.spec.js:41]: Okay, looking forward to that.

**Review by KufreOkon — APPROVED** [review]

**Pull request merged**